src/geo/map/cartodb-layer.js

```diff
@@ -129,6 +129,10 @@ var CartoDBLayer = LayerModelBase.extend({
 
   getSource: function () {
     return this.get('source');
+  },
+
+  setSource: function (source, options) {
+    this.set('source', source, options);
```

Comment:
I assume when the source is changed a `change` event is thrown and the visModel/Engine is listening in order to reload the visualization, right?